apps/system/js/app_usage_metrics.js

```diff
@@ -613,19 +613,76 @@
     };
 
     var deviceInfoQuery = {
-      'deviceinfo.update_channel': 'unknown',
-      'deviceinfo.platform_version': 'unknown',
+      'developer.menu.enabled': false, // If true, data is probably an outlier
+      'deviceinfo.hardware': 'unknown',
+      'deviceinfo.os': 'unknown',
       'deviceinfo.platform_build_id': 'unknown',
-      'developer.menu.enabled': false // If true, data is probably an outlier
+      'deviceinfo.platform_version': 'unknown',
+      'deviceinfo.product_model': 'unknown',
+      'deviceinfo.software': 'unknown',
+      'deviceinfo.update_channel': 'unknown'
     };
 
     // Query the settings db to get some more device-specific information
-    AUM.getSettings(deviceInfoQuery, function(deviceinfo) {
-      data.deviceinfo = deviceinfo;
+    AUM.getSettings(deviceInfoQuery, function(deviceInfo) {
+      data.deviceinfo = deviceInfo;
+      data.simInfo = getSIMInfo();
+
       // Now transmit the data
       send(data);
     });
 
+    function getSIMInfo() {
+      var simInfo = {
+        network: null,
+        icc: null
+      };
+
+      if (SIMSlotManager.noSIMCardConnectedToNetwork()) {
+        // No connected SIMs
+        return simInfo;
+      }
+
+      var slots = SIMSlotManager.getSlots().filter(function(slot) {
+        return !slot.isAbsent() && !slot.isLocked();
```

Comment:
I don't know anything about SIMSlotManager, but have you tested this on a dual sim device with only one card in it?  Do you get an array of two elements where one of the elements is null?  How about if the one card is in slot #2? Can that happen, and if so, do you need to put |slot &&| before the !slot.isAbsent()?